src/Events/SEO/No_Index.php

```diff
@@ -0,0 +1,204 @@
+<?php
+/**
+ * Functionality for interacting with the SEO nodindex.
+ *
+ * @since   6.1.0
+ *
+ * @package TEC\Events\SEO
+ */
+
+namespace TEC\Events\SEO;
+
+use \Tribe__Date_Utils as Dates;
+
+ /**
+  * Class Provider
+  *
+  * @since   6.1.0
+  * @package TEC\Events\Telemetry
+  */
+class No_Index {
```

Comment:
I would integrate this code, as suggested in a comment on the Service Provider, in a Controller and reduce the abstractions.
Also: test coverage?